src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityDAO.groovy

```diff
@@ -0,0 +1,212 @@
+package edu.oregonstate.mist.directoryapi
+
+import org.ldaptive.DefaultConnectionFactory
+import org.ldaptive.Connection
+import org.ldaptive.LdapEntry
+import org.ldaptive.LdapException
+import org.ldaptive.Response
+import org.ldaptive.SearchOperation
+import org.ldaptive.SearchRequest
+import org.ldaptive.SearchResult
+import org.ldaptive.pool.PooledConnectionFactory
+import org.ldaptive.pool.SoftLimitConnectionPool
+import org.ldaptive.pool.PoolConfig
+import java.util.regex.Pattern
+
+/**
+ * Directory entity data access object.
+ */
+class DirectoryEntityDAO {
+    private final String LDAP_URL
+    private final String BASE_DN
+    private final PooledConnectionFactory pooledConnectionFactory
+    private static final Pattern illegalCharacterPattern = Pattern.compile(
+            '''(?x)       # this extended regex defines
+               (?!        # any character that is not
+                  [
+                   a-zA-Z # a letter,
+                   0-9    # a number,
+                   -      # a hyphen,
+                   _      # an underscore,
+                   \\.    # a period, or
+                   @      # an at sign
+                  ])
+               .          # to be an illegal character.
+            ''')
+    private static final Pattern dollarSignPattern = Pattern.compile('\\$')
+
+    /**
+     * Constructs the directory entity data access object with given LDAP configuration.
+     *
+     * @param ldapConfiguration
+     */
+    public DirectoryEntityDAO(Map<String,Object> ldapConfiguration) {
```

Comment:
I don't believe it should be the responsibility of the DAO to manage the connection pool. The connection pool should be initialized by another class. The DAO should just be executing the LDAP queries.